### PR TITLE
Add Container Manager for execution result uploading, JSON test execution config

### PR DESCRIPTION
**Pull request by hafklin**
This PR primarily adds a `ContainerManager` that serves to start and monitor Docker containers for executors and uploads their results to a WebDAV server when finished. 
Furthermore, it extends the `ExecutorDescriptor` with a Results URL that specifies where the results of the test will be uploaded and allows this URL to be set in CLEO and LEA.
Lastly, it adds the possibility for a user to specify the test execution configuration in CLEO by means of a JSON file.

The executor parameters `volumes`, `devices`, and `ports` can still be set but are not considered when spawning a container. As I am not sure for which use cases those parameters were intended, we can maybe have a discussion about whether they are still needed.

The `ContainerManager` contains some basic functionality to constantly fetch logs from the running container. Currently, these logs are only printed using the `info!` macro. Properly handling these logs and making them accessible in Grafana in a nice way is still on my to-do list. 

This PR addresses several points in #5

### CODE REVIEW

**Comment by hafklin** [comment]
As agreed during our meeting yesterday, I implemented support for ports, devices, and volumes supplied to the container. Furthermore, I removed the dependency on the zip_extensions crate.
However, I did not change the check for the results canary file to a notify-based approach. The reason for this is that we do not only want to wait for the file, but we also want to regularly check the current state of the container, and we want to grab all logs from the `stdout` of `docker logs`. Therefore, I think that the current approach is likely best way to do this.

File: .ci/docker/nginx-webdav/localhost_ports.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+services:
+  nginx-webdav:
+    ports:
```
[the remark this answers is not in the document]
Comment (reimarstier) [reply .ci/docker/nginx-webdav/localhost_ports.yml:4]: i took the liberty of adding the file myself

**Review by reimarstier — COMMENTED** [review]
Apparently the large files flag produces a broken zip file. I took the liberty of adding a test to cover this and added a few other small things that I needed to test the setup of the webdav server.

File: opendut-edgar/src/service/test_execution/webdav_client.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    //         </D:propfind>
+    //     "#;
+
+    //     self.start_request(Method::from_bytes(b"PROPFIND").unwrap(), path)
+    //         .headers(self.custom_header("depth", depth))
+    //         .body(body)
+    //         .send()
+    //         .await
+    //         .map_err(|cause| Error::Request { method: String::from("PROPFIND"), cause } )
+    // }
```
[the remark this answers is not in the document]
Comment (hafklin) [reply opendut-edgar/src/service/test_execution/webdav_client.rs:71]: It probably won't be needed anymore, I removed it

File: .ci/deploy/localenv/data/nginx-webdav/docker-compose.yml
[the remark this answers is not in the document]
Comment (hafklin) [reply .ci/deploy/localenv/data/nginx-webdav/docker-compose.yml:1]: Oh, looks like I forgot to remove that. I did that now.

File: opendut-edgar/src/service/test_execution/container_manager.rs [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+        self.create_results_dir().await?;
+        self.start_container().await?;
+        self.log_reader = Some(
+            ContainerLogReader::create(
+                self.config.engine.to_string(), 
+                self.container_name.as_ref().unwrap().clone()
+            )?);
+
+        loop {
+            self.log_reader.as_mut().unwrap().read().await;
```
[the remark this answers is not in the document]
Comment (hafklin) [reply opendut-edgar/src/service/test_execution/container_manager.rs:83]: I agree, I also wasn't happy with how this looked. Thanks!

File: .ci/docker/nginx-webdav/localhost_ports.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hafklin) [reply .ci/docker/nginx-webdav/localhost_ports.yml:4]: > i took the liberty of adding the file myself

Thanks!

> Apparently the large files flag produces a broken zip file.
I'm surprised, I tested whether the zipped files created with the large file flag can be unzipped, which worked fine for me. But then I guess there are some issues. Without the flag, we cannot have results folders larger than 4GB, but I think that's an acceptable limitation for now.

**Comment by hafklin** [comment]
Thanks a lot for your reviews, @mbfm and @reimarstier. This was once again very helpful, I really appreciate it.
I incorporated all the changes you requested, so feel free to have a look at it again.

File: .ci/docker/nginx-webdav/localhost_ports.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (reimarstier) [reply .ci/docker/nginx-webdav/localhost_ports.yml:4]: I saw the `large_file` flag slipped back in. Was this intentional? How did you test it? On Ubuntu's unzip package (6.0-26ubuntu3.2), it does not work.

[the remark this answers is not in the document]
Comment (hafklin) [reply .ci/docker/nginx-webdav/localhost_ports.yml:4]: That's odd, I did not intentionally change it back. I'm not sure what went wrong there. Thanks for switching it back!
Back when I tested it, I used Ark (kali-rolling 4:23.08.5-1), with which unzipping worked fine. But if there are scenarios where it fails, I agree that we should keep it `false`.

[the remark this answers is not in the document]
Comment (reimarstier) [reply .ci/docker/nginx-webdav/localhost_ports.yml:4]: Ok, good to know.

**Review by reimarstier — APPROVED** [review]
LGTM

**Pull request merged**